.github/workflows/test.yaml

```diff
@@ -0,0 +1,55 @@
+name: Test
+
+on:
+  - pull_request
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+
+    env:
+      KEYSTORE_COMMON_NAME: keystore-${{ github.sha }}
+      TRUSTSTORE_COMMON_NAME: truststore-${{ github.sha }}
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: docker/setup-qemu-action@v3
+      - uses: docker/setup-buildx-action@v3
+      - uses: docker/build-push-action@v5
+        with:
+          context: .
+          tags: java-spiffe-helper:test
+          load: true
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+      - uses: helm/kind-action@v1
+        with:
+          cluster_name: kind
+      - run: kind load docker-image java-spiffe-helper:test --name kind
+      - run: helm upgrade --install -n spire-server spire-crds spire-crds --repo https://spiffe.github.io/helm-charts-hardened/ --create-namespace
+      - run: envsubst < .github/tests/spire-values.yaml.tpl > .github/tests/spire-values.yaml
```

Comment:
There's only two substitutions in the file, so could simplify it to just be static and override them in the helm install one line down adding:
  --set spire-server.ca_subject.common_name=$TRUSTSTORE_COMMON_NAME \
  --set spire-server.controllerManager.identities.clusterSPIFFEIDs.java-spiffe-helper.dnsNameTemplates[0]=$KEYSTORE_COMMON_NAME
  
  envsubst of a values file has always made me a little nervous.